tests/foreman/ui/test_registration.py

```diff
@@ -694,3 +697,58 @@ def test_global_registration_with_capsule_host(
     assert result.status == 0
     assert module_lce_library.name in result.stdout
     assert module_org.name in result.stdout
+
+
+@pytest.mark.tier2
+@pytest.mark.rhel_ver_match('[^6].*')
+def test_global_registration_upgrade_subscription_manager(
+    session, module_activation_key, module_os, rhel_contenthost
+):
+    """Host registration form produces a correct registration command and
+    subscription-manager can be updated from a custom repository before
+    registration is completed.
+
+    :id: b7a44f32-90b2-4fd6-b65b-5a3d2a5c5deb
+
+    :customerscenario: true
+
+    :expectedresults: Host is successfully registered, repo is enabled
+        on advanced tab and subscription-manager is updated.
+
+    :steps:
+        1. Create activation-key
+        2. Open the global registration form, add repo and activation key
+        3. Add 'subscription-manager' to install packages field
+        4. Check subscription-manager was installed from repo_name
+
+    :parametrized: yes
+
+    :BZ: 1923320
+    """
+    client = rhel_contenthost
+    repo_name = 'foreman_register'
+    rhel_ver = rhel_contenthost.os_version.major
+    repo_url = settings.repos.get(f'rhel{rhel_ver}_os')
+    if isinstance(repo_url, dict):
+        repo_url = repo_url['baseos']
+    # Ensure subs-man is installed from repo_name by removing existing package.
+    result = client.execute('rpm --erase --nodeps subscription-manager')
+    assert result.status == 0
```

Comment:
If the command failed, the RPM probably wasn't installed. Would that be fine, too?
```
$ rpm --erase --nodeps asoifsaogdsg
error: package asoifsaogdsg is not installed
$ echo $?
1
```

Reply:
I guess it should be fine if the command fails but do you think there can be any instance where subscription-manager is not installed?

Reply:
At this time, I doubt it. Hence my question, what do we check here? What if it fails, what does it mean? Does this assert really mean anything? I don't know, I have questions but no answers :)

Reply:
I don't think this assert would be needed. But I have no answer to what it means if this assertion fails.

Reply:
IIUC, here we're removing sub-man first and then installing it from foreman_register, but the test name says we're upgrading subman, so could you either update the test title or verify the upgraded version for subman package from foreman_register repo?

Reply:
@shweta83  this isn't updated yet? ^

Reply:
I have not added any check to see if the sub-man is updated because I don't know if it always going to be the case that the sub-man installed from foreman_register is always at least a version latest than  the existing one. If you check the BZ, the issue was about sub-man from foreman-register was not working so her we just need to uninstall and install it from repo. I have updated the test name and description though